lib/katello/permission_creator.rb

```diff
@@ -351,6 +351,11 @@ def subscription_permissions
                            'katello/api/v2/subscriptions' => [:delete_manifest]
                          },
                          :resource_type => 'Katello::Subscription'
+      @plugin.permission :manage_subscription_allocations,
+                         {
+                           'katello/api/v2/upstream_subscriptions' => [:index]
+                         },
+                         :resource_type => 'Katello::Subscription'
```

Comment:
If you add a new model object for the Upstream Pool I think you'll have to change this

Reply:
Not to my knowledge. This resource type is just what the permission is grouped under. afaiu, permissions are only controller#action based, not based on model or db access.